src/content/blog/legacy-projects-with-unsupported-php-and-mysql-using-ddev.md

```diff
@@ -0,0 +1,175 @@
+
+Just being able to "bait and switch" the PHP and DB container with a different base Docker image was all that was needed for me:
+
+## Step 1: Base config
+
+I created the base `~/legacyphp/.ddev/config.yaml` file manually inside my `~/legacyphp` project directory, setting `legacyphp` as the project name.
+
+Note that I configured PHP and MySQL versions that are supported by DDEV for this first:
+
+```
+name: legacyphp
+type: php
+docroot: htdocs
+php_version: "8.3"
+webserver_type: apache-fpm
+database:
+    type: mysql
+    version: "8.0"
+```
+
+## Step 2: Rewire DB
+
+Next I created the very small file `~/legacyphp/.ddev/docker-compose.db.yaml` in the same directory next to `config.yaml`:
+
+```
+services:
+  db:
+    platform: linux/amd64
+    build:
+      args:
+        BASE_IMAGE: ddev/ddev-dbserver-mysql-5.5:v1.24.6
+```
+
+Two things are noteworthy:
+
+- Setting `linux/amd64` as the platform will require Rosetta to be available on the macOS ARM64 platform
+- The `BASE_IMAGE` is set to a DDEV DB container of legacy ddev versions which are still available.
+
+## Step 3: Rewire PHP
+
+Using a different PHP work is just a few lines more work, because we are not replacing the whole `web` container oh DDEV, but instead add an additional PHP container which is executed from the web container via port 9000.
+
+This is done via the file `~/legacyphp/.ddev/docker-compose.php.yaml`:
+
+```
+services:
+  php:
+    container_name: ddev-${DDEV_SITENAME}-php
+    image: devilbox/php-fpm:5.3-work
+
+    restart: "no"
+    ports:
+      - 9000
+    labels:
+      com.ddev.site-name: ${DDEV_SITENAME}
+      com.ddev.approot: ${DDEV_APPROOT}
+    volumes:
+      - type: bind
+        source: ../
+        target: /var/www/html
+        consistency: cached
+      - ".:/mnt/ddev_config:ro"
+      - ddev-global-cache:/mnt/ddev-global-cache
+      - "./php:/etc/php-custom.d"
+    environment:
+      - DDEV_PHP_VERSION
+      - IS_DDEV_PROJECT=true
+  web:
+    links:
+    - php:php
+    healthcheck:
+      test: ["CMD", "true"]
+```
+
+Note here that we use `devilbox/php-fpm` with our needed version, and a bind-mount takes care the PHP container can access our main project root directory.
+
+A special mount of `~/legacyphp/.ddev/php/` is included so that we can control the `php.ini` configuration, if needed. For example you could disable the OPCache+APC in case you're doing some legacy benchmarking that should not be falsified via caching, I created a very small file `~/legacyphp/.ddev/php/php.ini` file with the contents:
+
+```
+# This is an example.
+# apc.enabled=Off
+# opcache.enable=Off
+```
+
+## Step 4: Utilize the PHP container with an Apache proxy
+
+To execute PHP with our external PHP docker image, I created the following file in `~/legacyphp/.ddev/apache/apache-site.conf`:
+
+```
```

Comment:
Not sure if there is something for apache.

Reply:
Ah. According to https://shiki.style/languages#bundled-languages `apache` should do the trick, will try.

Reply:
looks good to me, it has styling using that markup